### DISCUSSION TITLE
Why is shallow routing impossible between these routes, misunderstanding the caveat

### DISCUSSION THREAD

**Discussion by dan-cooke** [vercel/next.js#18789] [Help] [answered]
Very simple question:

With the following pages setup:

`pages/offense/index.js`

`pages/offense/[id]/index.js`

`pages/search/[...slug]/js`
         
Why does the following transition allow for shallow routing:

```
/offense/1 -> /offense
```

But the following does not

```
/offense/1 -> /search
```

The docs do not explain this [caveat](https://nextjs.org/docs/routing/shallow-routing#caveats) very clearly:

>Shallow routing only works for same page URL changes. For example, let's assume we have **another page** called pages/about.js, and you run this:
>`router.push('/?counter=10', '/about?counter=10', { shallow: true })`
> **Since that's a new page**, it'll unload the current page, load the new one and wait for data fetching even though we asked to do shallow routing.

I have highlighted the wording that does not make sense to me "**since that's a new page**"

But in the above scenario

`pages/offense/index.js` and `pages/offense/[id]/index.js` are different pages as well - but shallow routing is still allowed between them?

If I am misunderstanding something with the wording of this caveat, please let me know, and perhaps we could update the docs to be more clear

**Comment by a-m-dev** [comment]
+1

**Comment by lfades** [comment, marked as answer]
Shallow routing between `pages/offense/index.js` and `pages/offense/[id]/index.js` is not allowed, it shouldn't be possible for that case either because those are different pages.

**Comment by dan-cooke** [comment]
OK it appears that neither pages are being shallow routed and I'm just confusing the shallow routing for prefetching.
